### PR TITLE
Pass GitHub token to setup-r-dependencies in r-hub workflow

### PR DESCRIPTION
**Pull request by walkowif**

### CODE REVIEW

File: .github/workflows/rhub.yaml [hunk 1]
```diff
@@ -95,10 +95,11 @@ jobs:
           job-config: ${{ matrix.config.job-config }}
 
       - name: Prepare DESCRIPTION file
-        uses: insightsengineering/setup-r-dependencies@main
+        uses: insightsengineering/setup-r-dependencies@fix_internal_repo
```
Comment (walkowif) [inline .github/workflows/rhub.yaml:98]:
```suggestion
        uses: insightsengineering/setup-r-dependencies@main
```

File: .github/workflows/rhub.yaml [hunk 2]
```diff
@@ -145,10 +146,11 @@ jobs:
           job-config: ${{ matrix.config.job-config }}
 
       - name: Setup R dependencies
-        uses: insightsengineering/setup-r-dependencies@main
+        uses: insightsengineering/setup-r-dependencies@fix_internal_repo
```
Comment (walkowif) [inline .github/workflows/rhub.yaml:149]:
```suggestion
        uses: insightsengineering/setup-r-dependencies@main
```

File: .github/workflows/rhub.yaml (hunk 1 above)
Comment (walkowif) [reply .github/workflows/rhub.yaml:98]: Yes, good idea, we can tag the action after https://github.com/insightsengineering/setup-r-dependencies/pull/3 is merged.

Comment (pawelru) [reply .github/workflows/rhub.yaml:98]: Just merged this. You can now proceed with tagging and using tags here

Comment (walkowif) [reply .github/workflows/rhub.yaml]: `v1` and `v1.0.0` of `setup-r-dependencies` is now available [here](https://github.com/insightsengineering/setup-r-dependencies/releases/tag/v1.0.0).

**Review by pawelru — APPROVED** [review]
LGTM! 👍

**Pull request merged**